sklearn/neighbors/tests/test_neighbors.py

```diff
@@ -1325,3 +1327,97 @@ def test_pairwise_boolean_distance():
     nn1 = NN(metric="jaccard", algorithm='brute').fit(X)
     nn2 = NN(metric="jaccard", algorithm='ball_tree').fit(X)
     assert_array_equal(nn1.kneighbors(X)[0], nn2.kneighbors(X)[0])
+
+
+class MyRadiusNNAlgorithm(object):
+    def fit(self, X):
+        self._fitted = True
+
+    def query_radius(self, X, r, return_distance=True):
```

Comment:
This function is not using parameters passed in `fit` function. How will it compute points from fit function?
It would be okay of you describe the motivation behind this function and the below assert condition (other reviewed part)

Reply:
@thechargedneutron Sorry for the late reply. Yes, you are right it should take an X from fit. The idea was to prototype the simplest possible example of a custom NN search class with an API similar to KDTree.